cpp/tensorrt_llm/batch_manager/trtGptModelInflightBatching.cpp

```diff
@@ -555,17 +555,18 @@ std::shared_ptr<kv_cache_manager::KVCacheManager> TrtGptModelInflightBatching::c
     auto const tokensPerBlock = mModelConfig.getTokensPerBlock();
     auto const kvDtype = mModelConfig.getKvDataType();
 
-    bool enableCyclicKvCache = false;
+    bool enableSlidingWindowKvCache = false;
     for (SizeType32 maxAttenWin : getMaxAttentionWindowVec())
     {
         if (maxAttenWin != getMaxSequenceLen())
         {
-            enableCyclicKvCache = true;
+            enableSlidingWindowKvCache = true;
             break;
         }
     }
-    TLLM_CHECK_WITH_INFO(
-        getMaxBeamWidth() == 1 || !enableCyclicKvCache, "Can't support cyclic kv cache with beam search.");
+    // TODO (tasida): Should this be removed? Beam search + sliding window kv cache should be supported.
```

Comment:
I'm pretty sure beam search + sliding window attention is supported now, but wanted to validate with the reviewers before removing this assertion